numpy/core/include/numpy/ndarraytypes.h

```diff
@@ -1027,6 +1027,8 @@ typedef void (NpyIter_GetMultiIndexFunc)(NpyIter *iter,
  * eliminate overlap.
  */
 #define NPY_ITER_COPY_IF_OVERLAP            0x00002000
+/* At least one operand uses UPDATEIFCOPY semantics, use context manager */
+#define NPY_ITER_USE_CONTEXT_MANAGER        0x00004000
```

Comment:
As far as nditer is concerned, the context manager completely resolves any needed ``PyArray_ResolveUpdateIfCopy`` call without relying on refcount semantics or modifying ``NpyIter_Deallocate``

Now that I issue a warning when ``PyArray_ResolveUpdateIfCopy`` is called during deallocation, it turns out I was misreading the actual place needing conversion to a context manager. Since I did not convert all the nditer() calls to context managers, and there were places in tests that reuse the same variable name, triggering the deallocation, I actually needed to convert the *previous* test to a context manager.

Again, once I reworked the python-level tests, no warnings of a call to ``PyArray_ResolveUpdateIfCopy`` are emitted during nditer tests. However, there are a few places it seems ``PyArray_ResolveUpdateIfCopy`` is still needed, see my comment below